File: Beecrowd/Beginner/1435 - 1985/1827 - Square Array IV/Python/1827_SquareArrayIV.py
while True:
    try:
        array_size = int(input())
        matrix = [[0 for _ in range(array_size)] for _ in range(array_size)]

        for i in range(array_size):
            matrix[i][i] = 2
            matrix[i][array_size - i - 1] = 3

        middle = array_size // 3

        for i in range(middle, array_size - middle):
            for j in range(middle, array_size - middle):
                matrix[i][j] = 1

        matrix[array_size // 2][array_size // 2] = 4

        for row in matrix:
            print(''.join(map(str, row)))

        print()

    except EOFError:
        break

File: Beecrowd/Beginner/1435 - 1985/1827 - Square Array IV/C/1827_SquareArrayIV.c
#include <stdio.h>

int main() {
    int arraySize;

    while (scanf("%d", &arraySize) != EOF) {
        int matrix[arraySize][arraySize];

        for (int i = 0; i < arraySize; i++) {
            for (int j = 0; j < arraySize; j++) {
                matrix[i][j] = 0;
            }
        }

        for (int i = 0; i < arraySize; i++) {
            matrix[i][i] = 2;
            matrix[i][arraySize - i - 1] = 3;
        }

        int middle = arraySize / 3;

        for (int i = middle; i < arraySize - middle; i++) {
            for (int j = middle; j < arraySize - middle; j++) {
                matrix[i][j] = 1;
            }
        }

        matrix[arraySize / 2][arraySize / 2] = 4;

        for (int i = 0; i < arraySize; i++) {
            for (int j = 0; j < arraySize; j++) {
                printf("%d", matrix[i][j]);
            }

            printf("\n");
        }

        printf("\n");
    }

    return 0;
}
